### ISSUE TITLE
Pseudo dice always very small like 0.0015 and became 0 quickly.

### ISSUE THREAD

**Issue by Overflowu7** [MIC-DKFZ/nnUNet#1687]
My data have nice dice in nnUNet and I want to change something in the model witthout deepsupervison.BUT when I replace nnUNet default dice and model with my own dice and model,the Pseudo dice =0 appeared. I'm too confused about that.

I got some information:"Be sure to softmax first, then extract the second layer (i.e. the target layer), and do not change the input preds themselves, be sure to get a copy. Because there is an external evaluation mask that follows the same process, if you extract the second layer before sigmoid here, it will lead to an error in the external evaluation mask, and the psuedo dice will always be 0"
I change the dice with this idea ,But the problem still occurs. Here is my dice:
```
def my_get_dice_loss(preds: torch.tensor, target: torch.tensor):
  pred=torch.softmax(preds,dim=1)[:,1,:,:,:].unsqueeze(1)
  inter = (pred * target)
  union = (pred + target)
  # pred:BCHW, target: BCHW
  if (len(pred.shape) == 5) and (len(target.shape) == 5):
    inter = inter.sum(dim=2).sum(dim=2).sum(dim=2)
    union = union.sum(dim=2).sum(dim=2).sum(dim=2)
  dice_loss = 1 - 2 * (inter + 1) / (union + 2)
  # dice_loss = DiceCELoss(
  # to_onehot_y=False, softmax=False, squared_pred=False, smooth_nr=1e-5, smooth_dr=1e-5
  # )
  # result = dice_loss(pred, target)
  return dice_loss
```

**Comment by mrokuss** [comment]
Hey @Overflowu7 

It is a bit hard to judge what exactly goes wrong with the information provided. Generally I would suggest to take a look at the [implementation of the Dice Loss in nnUNet](https://github.com/MIC-DKFZ/nnUNet/blob/508f81dee7037a5beab1ff5b05c85a27741ed8bc/nnunetv2/training/loss/dice.py#L58) and just do your changes in a copy of this class. Or if you just want to use the Dice Loss without the CE Loss check out the [nnUNetTrainerDiceLoss](https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/training/nnUNetTrainer/variants/loss/nnUNetTrainerDiceLoss.py).

From this end, judging your code, I'd say it will not work for multiclass segmentation (since you only take the first channel) and expect the tensors to both be in one hot encoding already. Furthermore I am unsure if the formula you use for caluclating the dice loss is actually correct adding 1 on the top and 2 on the bottom. Hope this helps!

Max
